### PR TITLE
Uplift keys/dependencies to use more workspace inheritance

### PR DESCRIPTION
**Pull request by Jefffrey**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Try use workspace inheritance more, for both regular package keys, and also dependencies

Note that due to https://github.com/rust-lang/cargo/issues/11329 we need to specify `default-features = false` for the workspace dependency then override to `default-features = true` where we need, as the reverse does not work. If this isn't preferable we can revert back to the previous way of manually specifying these dependencies again without workspace inheritance.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -65,14 +65,14 @@ bytes = "1.4"
 chrono = { version = "0.4.34", default-features = false }
 ctor = "0.2.0"
 dashmap = "5.4.0"
-datafusion = { path = "datafusion/core", version = "36.0.0" }
-datafusion-common = { path = "datafusion/common", version = "36.0.0" }
+datafusion = { path = "datafusion/core", version = "36.0.0", default-features = false }
```
Comment (alamb) [inline Cargo.toml:68]: I wonder if we could add some comments (from the PR description) about why we need to use default-features=false here, and for only some of the crates

I am thinking about someone in the future who wants to make a new crate. How do they know if they need to set default features or not 🤔

**Review by alamb — APPROVED** [review]
This makes sense to me -- thank you @Jefffrey 

I wonder if there are any other tests we should run here to make sure it doesn't interfere with releases? @andygrove  any thoughts?

Otherwise I think we should merge up from main and then this PR would be good to go.

File: Cargo.toml (hunk 1 above)
Comment (Jefffrey) [reply Cargo.toml:73]: Good point, added a comment

File: Cargo.toml [hunk 2]
```diff
@@ -51,6 +51,11 @@ rust-version = "1.72"
 version = "36.0.0"
 
 [workspace.dependencies]
+# We turn off default-features for some dependencies here so the workspaces which inherit them can
```
Comment (alamb) [inline Cargo.toml:54]: 👍

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks again @Jefffrey

**Pull request merged**